crates/bevy_render/src/render_asset.rs

```diff
@@ -45,19 +49,112 @@ pub trait RenderAsset: Asset + Clone {
     ) -> Result<Self::PreparedAsset, PrepareAssetError<Self>>;
 }
 
-/// Whether or not to unload the [`RenderAsset`] after extracting it to the render world.
-///
-/// Unloading the asset saves on memory, as for most cases it is no longer necessary to keep
-/// it in RAM once it's been uploaded to the GPU's VRAM. However, this means you can no longer
-/// access the asset from the CPU (via the `Assets<T>` resource) once unloaded (without re-loading it).
-///
-/// If you never need access to the asset from the CPU past the first frame it's loaded on,
-/// or only need very infrequent access, then set this to Unload. Otherwise, set this to Keep.
-#[derive(Reflect, Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Default, Debug)]
-pub enum RenderAssetPersistencePolicy {
-    Unload,
-    #[default]
-    Keep,
+bitflags::bitflags! {
+    /// Defines where the asset will be used.
+    ///
+    /// If an asset is set to the `RENDER_WORLD` but not the `MAIN_WORLD`, the asset will be
+    /// unloaded from the asset server once it's been extracted and prepared in the render world.
+    ///
+    /// Unloading the asset saves on memory, as for most cases it is no longer necessary to keep
+    /// it in RAM once it's been uploaded to the GPU's VRAM. However, this means you can no longer
+    /// access the asset from the CPU (via the `Assets<T>` resource) once unloaded (without re-loading it).
+    ///
+    /// If you never need access to the asset from the CPU past the first frame it's loaded on,
+    /// or only need very infrequent access, then set this to `RENDER_WORLD`. Otherwise, set this to
+    /// `RENDER_WORLD | MAIN_WORLD`.
+    ///
+    /// If you have an asset that doesn't actually need to end up in the render world, like an Image
+    /// that will be decoded into another Image asset, use `MAIN_WORLD`.
+    #[repr(transparent)]
+    #[derive(Serialize, TypePath, Deserialize, Hash, Clone, Copy, PartialEq, Eq, Debug)]
+    pub struct RenderAssetUsages: u8 {
```

Comment:
We might want to just use the bitflags crate here.

Reply:
This is wrapped in `bitflags::bitflags! { ... }`